api-management.adoc

```diff
@@ -0,0 +1,67 @@
+// JBoss, Home of Professional Open Source
+// Copyright 2016, Red Hat, Inc. and/or its affiliates, and individual
+// contributors by the @authors tag. See the copyright.txt in the
+// distribution for a full listing of individual contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+### Use API Management to take control of your microservices
+
+APIcast (https://github.com/3scale/apicast) is an Open Source API Gateway whose main focus areas are high performance and extensibility. It is part of the Redhat 3scale API Management solution, and is used by hundreds of companies around the world to expose their APIs in a secure and controlled way.
+
+#### Create a custom API Management build
+
+----
+$ git clone https://github.com/redhat-helloworld-msa/api-management
+$ cd api-management/
+$ oc new-build --binary --name api-management -e BACKEND_URL=http://127.0.0.1:8081
+----
+
+##### Specify the OpenShift domain
+
+----
+$ oc env bc/api-management OS_SUBDOMAIN=<OPENSHIFT-DOMAIN>
+
+#Using CDK3
+$ oc env bc/api-management OS_SUBDOMAIN=$(minishift ip).nip.io
+
+#Example: OS_SUBDOMAIN=192.168.64.11.nip.io
+$ oc env bc/api-management OS_SUBDOMAIN=192.168.64.11.nip.io
+----
+
+#### Create a custom API Management instance
+
+----
+$ oc start-build api-management --from-dir=. --follow
+$ oc new-app api-management
+$ oc expose svc/api-management --name api-bonjour
+$ oc expose svc/api-management --name api-hola
+$ oc expose svc/api-management --name api-ola
+$ oc expose svc/api-management --name api-aloha
+----
+
+##### (Optional) Enable Readiness probe
+
+----
+$ oc set probe dc/api-management --readiness --get-url=http://:8081/status/ready
+----
+
+#### Tell frontend to enable API Management
+
+----
+# Using CDK3
+$ oc env dc KEYCLOAK_AUTH_SERVER_URL=http://keycloak-sso.`minishift ip`.nip.io/auth -l app
```

Comment:
Because SSO use case is required, this instruction should be already made during the installation of SSO. If this is specific to this application we should use "oc env dc/api-management KEYCLOAK_AUTH_SERVER_URL=http://keycloak-sso.`minishift ip`.nip.io/auth"

Reply:
done, removed as this is done in the SSO configuration.